spectrum-cardano-connector/src/rocksdb.rs

```diff
@@ -0,0 +1,474 @@
+
+                        db_tx.delete(&best_block_key).unwrap();
+
+                        // The new best block will now be the parent of the old best block, if the parent
+                        // exists in the cache.
+                        if db_tx
+                            .get(&postfixed_key(&parent_id, PARENT_POSTFIX))
+                            .unwrap()
+                            .is_some()
+                        {
+                            let parent_id_slot_bytes = db_tx
+                                .get(&postfixed_key(&parent_id, SLOT_POSTFIX))
+                                .unwrap()
+                                .unwrap();
+                            let parent_id_slot = bincode::deserialize(&parent_id_slot_bytes).unwrap();
+                            let parent_block_number_bytes = db_tx
+                                .get(&postfixed_key(&parent_id, BLOCK_NUMBER_POSTFIX))
+                                .unwrap()
+                                .unwrap();
+                            let parent_block_number =
+                                bincode::deserialize(&parent_block_number_bytes).unwrap();
+
+                            db_tx
+                                .put(
+                                    &best_block_key,
+                                    bincode::serialize(&BlockRecord {
+                                        id: parent_id,
+                                        slot: parent_id_slot,
+                                        block_number: parent_block_number,
+                                    })
+                                    .unwrap(),
+                                )
+                                .unwrap();
+                        }
+                        match db_tx.commit() {
+                            Ok(_) => {
+                                return Some(Block {
+                                    id,
+                                    parent_id,
+                                    slot,
+                                    block_number,
+                                    transactions,
+                                });
+                            }
+                            Err(e) => {
+                                if e.kind() == rocksdb::ErrorKind::Busy {
+                                    continue;
+                                } else {
+                                    panic!("Unexpected error: {}", e);
+                                }
+                            }
+                        }
+                    } else {
+                        return None;
+                    }
+                } else {
+                    println!("NO BEST BLOCK!");
+                    return None;
+                }
+            }
+        })
+        .await
+    }
+}
+
+#[derive(Clone)]
+pub struct Block {
+    pub id: Hash<32>,
+    pub parent_id: Hash<32>,
+    pub slot: u64,
+    pub block_number: u64,
```

Comment:
Can't use slot number to measure maximum rollback length. The block number here is used, but I'm not sure exactly what it represents (I searched the cardano docs but no luck). It's definitely not height from genesis block though.